### PR TITLE
feat: update dependency permission

### PR DESCRIPTION
**Pull request by kwasniew**
## About the changes

* adding new project level permission for updating dependencies (adding, editing, removing) called "Update feature dependency"
* project owner and members are given this by default

Screenshot 2023-10-02 at 15 46 45

#### Add without permission
Screenshot 2023-10-02 at 15 48 07

#### Add with permission (had to remove plus icon since it breaks the without permission state with lock - only one icon is possible)
Screenshot 2023-10-02 at 15 47 50

#### Edit/delete without permission:
Screenshot 2023-10-02 at 15 47 14

#### Edit/delete with permission:
Screenshot 2023-10-02 at 15 47 42

### CODE REVIEW

File: frontend/src/component/feature/FeatureView/FeatureOverview/FeatureOverviewSidePanel/FeatureOverviewSidePanelDetails/DependencyRow.tsx [hunk 1]
```diff
@@ -30,14 +32,16 @@ export const DependencyRow: FC<{ feature: IFeatureToggle }> = ({ feature }) => {
                     <FlexRow>
                         <StyledDetail>
                             <StyledLabel>Dependency:</StyledLabel>
-                            <Button
-                                startIcon={<Add />}
```
Comment (kwasniew) [inline frontend/src/component/feature/FeatureView/FeatureOverview/FeatureOverviewSidePanel/FeatureOverviewSidePanelDetails/DependencyRow.tsx:34]: startIcon was breaking permission button disabled state

File: frontend/src/component/providers/AccessProvider/permissions.ts [hunk 2]
```diff
@@ -1,6 +1,7 @@
 export const ADMIN = 'ADMIN';
 export const CREATE_FEATURE = 'CREATE_FEATURE';
 export const UPDATE_FEATURE = 'UPDATE_FEATURE';
+export const UPDATE_DEPENDENCY = 'UPDATE_DEPENDENCY';
```
Comment (ivarconr) [inline frontend/src/component/providers/AccessProvider/permissions.ts:4]: can we call it `UPDATE_FEATURE_DEPENDENCY`?

Comment (kwasniew) [reply frontend/src/component/providers/AccessProvider/permissions.ts]: sure, updating across all files to keep it consistent.

File: src/migrations/20231002122426-update-dependency-permission.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
+exports.up = function (db, cb) {
+    db.runSql(
+        `
+        INSERT INTO permissions (permission, display_name, type) VALUES ('UPDATE_FEATURE_DEPENDENCY', 'Update feature dependency', 'project');
+        SELECT assign_unleash_permission_to_role('UPDATE_FEATURE_DEPENDENCY', 'Member');
+        SELECT assign_unleash_permission_to_role('UPDATE_FEATURE_DEPENDENCY', 'Owner');
```
Comment (FredrikOseberg) [inline src/migrations/20231002122426-update-dependency-permission.js:6]: Cool, I didn't know you could do this

**Review by FredrikOseberg — APPROVED** [review]
LGTM

**Pull request merged**